users/bcat/bcat_oled.c

```diff
@@ -0,0 +1,221 @@
+                .wpm  = get_current_wpm(),
+            };
+            oled_task_keymap(&keyboard_state);
+        } else if (is_oled_on()) {
+            oled_off();
+        }
+#if defined(SPLIT_KEYBOARD)
+    } else {
+        /* Display logo embedded at standard location in the OLED font on the
+         * slave side. By default, this is a "QMK firmware" logo, but many
+         * keyboards substitute their own logo. Occupies 21x3 character cells.
+         *
+         * Since the slave display buffer never changes, we don't need to worry
+         * about oled_render incorrectly turning the OLED on. Instead, we rely
+         * on SPLIT_OLED_ENABLE to propagate OLED on/off status from master.
+         */
+        static const char PROGMEM logo[] = {
+            // clang-format off
+            0x80, 0x81, 0x82, 0x83, 0x84, 0x85, 0x86, 0x87, 0x88, 0x89, 0x8a, 0x8b, 0x8c, 0x8d, 0x8e, 0x8f, 0x90, 0x91, 0x92, 0x93, 0x94,
+            0xa0, 0xa1, 0xa2, 0xa3, 0xa4, 0xa5, 0xa6, 0xa7, 0xa8, 0xa9, 0xaa, 0xab, 0xac, 0xad, 0xae, 0xaf, 0xb0, 0xb1, 0xb2, 0xb3, 0xb4,
+            0xc0, 0xc1, 0xc2, 0xc3, 0xc4, 0xc5, 0xc6, 0xc7, 0xc8, 0xc9, 0xca, 0xcb, 0xcc, 0xcd, 0xce, 0xcf, 0xd0, 0xd1, 0xd2, 0xd3, 0xd4,
+            0x00,
+            // clang-format on
+        };
+
+        oled_write_P(logo, /*invert=*/false);
+    }
+#endif
+}
+
+void render_oled_layers(void) {
+    oled_advance_char();
+    oled_advance_char();
+#if defined(BCAT_ORTHO_LAYERS)
+    oled_write_char(IS_LAYER_ON(LAYER_LOWER) ? TRIANGLE_DOWN : ' ', /*invert=*/false);
+    oled_advance_char();
+    oled_write_char(IS_LAYER_ON(LAYER_RAISE) ? TRIANGLE_UP : ' ', /*invert=*/false);
+#else
+    switch (get_highest_layer(layer_state)) {
+        case LAYER_FUNCTION_1:
+            oled_write_P(PSTR("FN1"), /*invert=*/false);
+            break;
+        case LAYER_FUNCTION_2:
+            oled_write_P(PSTR("FN2"), /*invert=*/false);
+            break;
+        default:
+            oled_write_P(PSTR("   "), /*invert=*/false);
+            break;
+    }
+#endif
+}
+
+void render_oled_indicators(led_t leds) {
+    oled_advance_char();
+    oled_advance_char();
+    oled_write_P(leds.num_lock ? PSTR("NUM") : PSTR("   "), /*invert=*/false);
+    oled_advance_char();
+    oled_advance_char();
+    oled_write_P(leds.caps_lock ? PSTR("CAP") : PSTR("   "), /*invert=*/false);
+    oled_advance_char();
+    oled_advance_char();
+    oled_write_P(leds.scroll_lock ? PSTR("SCR") : PSTR("   "), /*invert=*/false);
+}
+
+void render_oled_wpm(uint8_t wpm) {
+    static const uint16_t UPDATE_MILLIS  = 100;
+    static uint32_t       update_timeout = 0;
+
+    if (timer_expired32(timer_read32(), update_timeout)) {
+        char wpm_str[] = "   ";
+        if (wpm > 0) {
+            wpm_str[2] = '0' + wpm % 10;
+        }
+        if (wpm >= 10) {
+            wpm_str[1] = '0' + wpm / 10 % 10;
+        }
+        if (wpm >= 100) {
+            wpm_str[0] = '0' + wpm / 100 % 10;
+        }
```

Comment:
This can be replaced with `get_u8_str(wpm, ' ')`, now.

Reply:
Thanks, done!